clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -708,6 +745,28 @@ protected Node curNode() {
             return curNode;
         }
 
+        public RetryContext getRetryContext() {
+            return retryContext;
+        }
+
+        public void retry(Call failedCall, Throwable throwable) {
+            long now = time.milliseconds();
+            RetryContext failedCallRetryContext = failedCall.getRetryContext();
+            this.retryContext.update(failedCallRetryContext);
+
+            if (this.retryContext.getTries() > maxRetries) {
+                failWithTimeout(now, throwable);
+                return;
+            }
+
+            if (log.isDebugEnabled()) {
```

Comment:
It's a bit redundant to check for `log.isDebugEnabled()` if we are just logging it at debug right?

Reply:
Yeah make sense. Remove the condition.